### PR TITLE
feature/time-refactor

### PR DESCRIPTION
**Pull request by emiax**
- Change `Time` class to become a non-singleton
- Move ownership of the current time to `TimeManager `(instead of singleton access).
- Store the `Time` as a `Syncable` in `TimeManager` instead of representing all member variables of `Time 
` as `Syncables`.
- Pass a Time object around in the update/render methods, so that renderables don't have to query the `OpenSpaceEngine `to know if time is paused or if it jumped.
- Introduce Timeline and Keyframe classes
- Make use of `Timeline `and `Keyframe `classes in `KeyframeInteractionMode` and `TimeManager`
- Added basic unit tests for `Timeline `and `Keyframe`

Future work: Add interpolation schemes for keyframes. Possibly use keyframes+interpolation feature to tween/morph properties, or figure out if this should be a separate mechanism.

### CODE REVIEW

File: include/openspace/util/timemanager.h [hunk 1]
```diff
@@ -28,23 +28,26 @@
...
-    void addKeyframe(const datamessagestructures::TimeKeyframe& kf);
+    void addKeyframe(double timestamp, Time kf);
     void removeKeyframesBefore(double timestamp);
     void removeKeyframesAfter(double timestamp);
     void clearKeyframes();
-    const std::deque<datamessagestructures::TimeKeyframe>& keyframes() const;
+    size_t nKeyframes() const;
 private:
+    Timeline<Time> _timeline;
+	SyncData<Time> _currentTime;
```
Comment (kbladin) [inline include/openspace/util/timemanager.h]: Indentation!!!!!!!!!!

File: include/openspace/util/timemanager.h [hunk 2]
```diff
@@ -28,23 +28,26 @@
 #include <vector>
 #include <deque>
 #include <openspace/network/messagestructures.h>
```
Comment (kbladin) [inline include/openspace/util/timemanager.h]: Remove include

File: src/network/parallelconnection.cpp [hunk 3]
```diff
@@ -1048,10 +1060,12 @@ void ParallelConnection::sendTimeKeyframe() {
     // Create a keyframe with current position and orientation of camera
     datamessagestructures::TimeKeyframe kf;
     
-    kf._dt = Time::ref().deltaTime();
-    kf._paused = Time::ref().paused();
+	Time& time = OsEng.timeManager().time();
```
Comment (kbladin) [inline src/network/parallelconnection.cpp]: Indentation

File: include/openspace/util/timeline.h [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+#include <deque>
+#include <cstddef>
+
+namespace openspace {
+
+/**
+* Base class for keyframes
+*/
+struct KeyframeBase {
+    KeyframeBase(size_t i, double t)
```
Comment (alexanderbock) [inline include/openspace/util/timeline.h]: The constructor definition is not necessary since it is a trivial ctor

**Pull request merged**